source/PlayerInfo.cpp

```diff
@@ -1043,7 +1043,7 @@ void PlayerInfo::SetFlagship(Ship &other)
 	for(const shared_ptr<Ship> &ship : ships)
 	{
 		bool shouldFollowFlagship = (ship != flagship && !ship->IsParked() &&
-			(!ship->CanBeCarried() || ship->JumpNavigation().JumpFuel()));
+			(!ship->CanBeCarried() || ship->JumpNavigation().JumpFuel() || flagship->CanBeCarried()));
```

Comment:
Doesn't this break if you have a carrier as an escort? Presumably you'd want the fighter escorts to dock to the carrier escort?

Reply:
They still do that, as iirc there is a later step in the departing process where fighters get assigned to carriers.

Reply:
`PlayerInfo::TakeOff` does indeed assign carryable ships appropriate carriers as parents. So, this is probably fine.

However, I think it would be better to modify that stage to handle any carryable ships that are left without a parent.
Even if your flagship can't be carried, if you have more carryable ships than you have bays, some will still be left parentless at take off. They will not be given a parent until `AI::Step`.
This doesn't take very long for player ships, but it's still not great to have player ships launch parentless.

Alternatively, just assign the flagship as the parent for every ship at this point, regardless of carryability of the escort or the flagship. Then, any carryable ships that can be assigned to other escorts with bays will be in `PlayerInfo::TakeOff`, any that can't will continue to have the flagship as their parent.

I think I actually like this last one the best.